### PR TITLE
Added signing plugins section to plugins topic

### PR DESCRIPTION
**Pull request by justinegeffen**
As per: https://github.com/mattermost/mattermost-developer-documentation/pull/360, content was added to this document.

### CODE REVIEW

File: source/administration/plugins.rst [hunk 1]
```diff
@@ -77,12 +77,163 @@ Plugins may have one or both of the following parts:
...
+To delete public key(s) from your Mattermost server, use:
+
+.. code-block:: sh
+
+  mattermost plugin delete key my-pk-file1 my-pk-file2
+
+Implementation
+~~~~~~~~~~~~~~~~~
+
+See the _`implementation document <https://docs.google.com/document/d/1qABE7VEx4k_ZAeh6Ydn4pGbu6BQfZt65x68i2s65MOQ/>`_for more information.
```
Comment (justinegeffen) [inline source/administration/plugins.rst]: Is this an appropriate link to include? Should we rather leave this out?

File: source/administration/plugins.rst [hunk 2]
```diff
@@ -77,12 +77,163 @@ Plugins may have one or both of the following parts:
...
 While this power enables deep customization and integration, it can be abused in the wrong hands. Plugins have full access to your server configuration and thus also to your Mattermost database. Plugins can read any message in any channel, or perform any action on behalf of any user in the webapp.
 
 You should only install custom plugins from sources you trust to avoid compromising the security of your installation.
 
+Plugin Signing
+---------------
+
+The Plugin Marketplace allows system administrators to download and install plugins from a central repository. Plugins installed via the Plugin Marketplace must be signed by a public key certificate trusted by the local Mattermost server.
+
+While the server ships with a default certificate used to verify plugins from the default Mattermost plugin marketplace, the server can be configured to trust different certificates and point at a different plugin marketplace. This document outlines the steps for generating a public key certificate and signing plugins for use with a custom plugin marketplace. It assumes access to the _`GNU Privacy Guard (GPG)]<https://gnupg.org>`_ tool.
```
Comment (amyblais) [inline source/administration/plugins.rst]:
```suggestion
While the server ships with a default certificate used to verify plugins from the default Mattermost plugin marketplace, the server can be configured to trust different certificates and point at a different plugin marketplace. This document outlines the steps for generating a public key certificate and signing plugins for use with a custom plugin marketplace. It assumes access to the _`GNU Privacy Guard (GPG)<https://gnupg.org>`_ tool.
```

File: source/administration/plugins.rst [hunk 3]
```diff
@@ -77,12 +77,163 @@ Plugins may have one or both of the following parts:
...
+.. code-block:: sh
+
+   PluginSettings.RequirePluginSignature = true
+
+is used to enforce plugin signature verification. With flag on, only marketplace plugins will be installed and verified. With flag off, customers will be able to install plugins manually without signature verification. Note that the marketplace plugins will still be verified even if flag is off.
+
+Key Generation
+~~~~~~~~~~~~~~~
+
+  Public and private key pairs are needed to sign and verify plugins. The private key is used for signing and should be kept in a secure location. The public key is used for verification and can be distributed freely. To generate a key pair, run the following command:
```
Comment (amyblais) [inline source/administration/plugins.rst]:
```suggestion
Public and private key pairs are needed to sign and verify plugins. The private key is used for signing and should be kept in a secure location. The public key is used for verification and can be distributed freely. To generate a key pair, run the following command:
```

File: source/administration/plugins.rst [hunk 4]
```diff
@@ -77,12 +77,163 @@ Plugins may have one or both of the following parts:
...
+To delete public key(s) from your Mattermost server, use:
+
+.. code-block:: sh
+
+  mattermost plugin delete key my-pk-file1 my-pk-file2
+
+Implementation
+~~~~~~~~~~~~~~~~~
+
+See the _`implementation document <https://docs.google.com/document/d/1qABE7VEx4k_ZAeh6Ydn4pGbu6BQfZt65x68i2s65MOQ/>`_for more information.
```
Comment (amyblais) [inline source/administration/plugins.rst]:
```suggestion
See the _`implementation document <https://docs.google.com/document/d/1qABE7VEx4k_ZAeh6Ydn4pGbu6BQfZt65x68i2s65MOQ/>`_ for more information.
```

File: source/administration/plugins.rst (hunk 2 above)
Comment (justinegeffen) [reply source/administration/plugins.rst]: Ah, that slipped past me. Thanks, @amyblais.

File: source/administration/plugins.rst [hunk 5]
```diff
@@ -77,12 +77,163 @@ Plugins may have one or both of the following parts:
...
 While this power enables deep customization and integration, it can be abused in the wrong hands. Plugins have full access to your server configuration and thus also to your Mattermost database. Plugins can read any message in any channel, or perform any action on behalf of any user in the webapp.
 
 You should only install custom plugins from sources you trust to avoid compromising the security of your installation.
 
+Plugin Signing
+---------------
+
+The Plugin Marketplace allows system administrators to download and install plugins from a central repository. Plugins installed via the Plugin Marketplace must be signed by a public key certificate trusted by the local Mattermost server.
+
+While the server ships with a default certificate used to verify plugins from the default Mattermost plugin marketplace, the server can be configured to trust different certificates and point at a different plugin marketplace. This document outlines the steps for generating a public key certificate and signing plugins for use with a custom plugin marketplace. It assumes access to the _`GNU Privacy Guard (GPG)<https://gnupg.org>`_ tool.
```
Comment (amyblais) [inline source/administration/plugins.rst]:
```suggestion
While the server ships with a default certificate used to verify plugins from the default Mattermost plugin marketplace, the server can be configured to trust different certificates and point at a different plugin marketplace. This document outlines the steps for generating a public key certificate and signing plugins for use with a custom plugin marketplace. It assumes access to the _`GNU Privacy Guard (GPG) <https://gnupg.org>`_ tool.
```

Comment (amyblais) [reply source/administration/plugins.rst]: The link is still not working

File: source/administration/plugins.rst [hunk 6]
```diff
@@ -77,12 +77,163 @@ Plugins may have one or both of the following parts:
...
+To delete public key(s) from your Mattermost server, use:
+
+.. code-block:: sh
+
+  mattermost plugin delete key my-pk-file1 my-pk-file2
+
+Implementation
+~~~~~~~~~~~~~~~~~
+
+See the _`implementation document <https://docs.google.com/document/d/1qABE7VEx4k_ZAeh6Ydn4pGbu6BQfZt65x68i2s65MOQ/>`_ for more information.
```
Comment (amyblais) [inline source/administration/plugins.rst]: The link is not formatted correctly yet - let me know if you need help.

Comment (justinegeffen) [reply source/administration/plugins.rst]: Thanks, @amyblais. Looks like I fixed it now.

File: source/administration/plugins.rst (hunk 5 above)
Comment (justinegeffen) [reply source/administration/plugins.rst]: Fixed!

**Pull request merged**